app/Actions/Photo/Strategies/AddStandaloneStrategy.php

```diff
@@ -171,7 +172,7 @@ protected function normalizeOrientation(): void
 			// and the checksum may have changed.
 			/* @var  Extractor $metadataExtractor */
 			$metadataExtractor = resolve(Extractor::class);
-			$this->photo->filesize = $metadataExtractor->filesize($absolutePath);
+			$this->photo->size_variants->getOriginal()->filesize = $metadataExtractor->filesize($absolutePath);
```

Comment:
As I will probably say again later, I feel sorry to propose poor quality code. I thought that I sufficiently tested and understood the code, but I clearly did not take enough time to understand consequences of the code.

I did not test the auto-rotation feature (didn't know it) and obviously it throws an error.

Anyway, I will remove this line because the size variant filesize is set in the variant factory.

Thanks for your vigilance.

Reply:
Au contraire, I honestly think that you delivered high-quality code. In particular, you used type annotations and you added comments to your code. Both aspects are not a matter of course. Moreover, your comments were helpful and provided the reader (and future programmer) with helpful insights. Unfortunately, I have read too many poor comments in my life which only repeated information which a reader could also deduce from the code itself, but completely missed to comment on the why and how the code relates to the rest of the program. Your comments don't fall into that category and this is a rather rare occasion. 👍

Reply:
Thanks for your kind words. 

See 0a54f0eb.

Reply:
However @nagmat84  I have a really bizarre phenomenon with auto resize and filesizes. I cannot think to an obvious reason, do you have any hints ?

1. Take `mongolia.jpeg` example image and fake orientation :
```
exiftool -Orientation=8 mongolia.jpeg
```
2. Import it in Lychee : auto-rotation takes place.
3. Check values for filesize in DB :
```
sqlite> select * from size_variants;
44|LOXWvW3mP85ugGUVoaas_fuy|0|big/97c0d9b2b4e2b17c0a41e34eb1099167.jpeg|1280|850|201315
45|LOXWvW3mP85ugGUVoaas_fuy|6|thumb/97c0d9b2b4e2b17c0a41e34eb1099167.jpeg|200|200|14800
46|LOXWvW3mP85ugGUVoaas_fuy|5|thumb/97c0d9b2b4e2b17c0a41e34eb1099167@2x.jpeg|400|400|45564
47|LOXWvW3mP85ugGUVoaas_fuy|4|small/97c0d9b2b4e2b17c0a41e34eb1099167.jpeg|542|360|51115
48|LOXWvW3mP85ugGUVoaas_fuy|3|small/97c0d9b2b4e2b17c0a41e34eb1099167@2x.jpeg|1084|720|160063
```
4. Drop all values :
```
sqlite> update size_variants set filesize=-1;
```
5. Run the command to set filesizes :
```
php artisan lychee:variant_filesize
```
6. Re-check filesizes values :
```
sqlite> select * from size_variants;
44|LOXWvW3mP85ugGUVoaas_fuy|0|big/97c0d9b2b4e2b17c0a41e34eb1099167.jpeg|1280|850|203237
45|LOXWvW3mP85ugGUVoaas_fuy|6|thumb/97c0d9b2b4e2b17c0a41e34eb1099167.jpeg|200|200|14800
46|LOXWvW3mP85ugGUVoaas_fuy|5|thumb/97c0d9b2b4e2b17c0a41e34eb1099167@2x.jpeg|400|400|45564
47|LOXWvW3mP85ugGUVoaas_fuy|4|small/97c0d9b2b4e2b17c0a41e34eb1099167.jpeg|542|360|51115
48|LOXWvW3mP85ugGUVoaas_fuy|3|small/97c0d9b2b4e2b17c0a41e34eb1099167@2x.jpeg|1084|720|160063
```

Notice how the original variant filesize is a little bit different.

Reply:
No, not by heart, but I have an educated guess. PHP caches access to files and there meta-data internally (i.e. the low-level commands `filesize`, `is_file`, `file_exist`, etc.) don't access the underlying filesystem, if there was a previous call for the same file and PHP finds the result in its cache.

As the metadata extractor calls `filesize` before we auto-rotate the file and then calls `filesize` again, we might get the (wrong, outdated) result from before the rotation.

However, if you nuke the entries in the DB and then call the console command, this is another PHP process with an empty cache.

Reply:
For example, see here: https://stackoverflow.com/questions/3747982/php-filesize-reporting-old-size

Reply:
Nice guess. However, running `clearstatcache()`, even globally (i.e. without a file path) after the image has been rotated (and before variant creations, according to your first comment) does not change the behaviour. 

I noticed that the same thing occurs, but only with 1 byte difference, on some image without auto-rotation. 

This may be worth investigating, as I guess it was already behaving that way before my modifications, and could mislead programs depending on it (not much, for now...) ?

Reply:
Again you were right, clearing the cache was needed as the path do not changes.
Also, simply removing the line was not enough as the variant creation code was using the old size.

Could you please review 3c0a33bae61deba0ef51d9e863b2290a53528c1c ?

Reply:
Looks good, although a little bit hack-ish, but again this is not your fault but due to Lychee's code which has already been there.

IHMO, in an optimal world, we would use `FlysystemFile` everywhere and the image handlers (i.e. `GdHandler` and `ImagickHandler`) would take care of flushing the case when they modify an image file. It should be the concern of the code which modifies data to invalidate cached data, not the concern of the caller.

But we should leave it like this for now. From my point of view, this conversation can be resolved.